tools/invoke-tests

```diff
@@ -51,6 +51,24 @@ fi
 export OS_AUTOINST_BUILD_DIRECTORY=$build_directory
 export OS_AUTOINST_MAKE_TOOL=$make_path
 
-# invoke tests within the t and xt directory
-cd t && "$prove_path" $PROVE_ARGS ${TESTS:-'-r'}
-cd ../xt && "$prove_path" $PROVE_ARGS ${TESTS:-'-r'}
+# prepare test arguments for the t directory
+cd t
+argt=""
+for word in ${TESTS:-'-r'}
+do
+    if [[ $(ls | grep -x $word) ]]; then argt="${argt}$word "; fi
```

Comment:
Calling `ls | grep …` in a loop seems bad performance-wise. The code also seems wrong because `argt` is only set but never used.

Maybe it makes more sense if the caller specifies the subdirectory as well, e.g. `TESTS='xt/foo.t t/bar.t'`. This would at least ease the manual splitting a little bit. Or we simply require two different invocations of this script and the correct directory needs to be passed as CLI argument (maybe defaulting to `t`). The required changes on the CMake-side shouldn't be too challenging (at least better than this).

Reply:
Nice catch, corrected. This pr is complete now, at least as far as the initial implementation is concerned, so I'm removing the WIP.

As for the suggestions: the first one is trivial. I considered that the exact original intended behavior a requirement, thus the current implementation . The second one would not let the user run tests in both directories in a single run, should they want to, and wouldn't be my first choice.

If you agree, I have the implementation of your first suggestion ready. The change should also be reflected in the documentation. I'll be waiting for the ok to push it.

Reply:
Nice catch, corrected. This pr is complete now, at least as far as the initial implementation is concerned.

As for the suggestions: the first one is trivial. I considered that the exact original intended behavior a requirement, thus the current implementation . The second one would not let the user run tests in both directories in a single run, should they want to, and wouldn't be my first choice.

Since @perlpunk agrees as well, I will implement your first suggestion.

Reply:
> I considered that the exact original intended behavior a requirement, thus the current implementation .

I only created this script to avoid redundancy between the autotools-based and CMake-based build script when porting to CMake. It is not a requirement that this script behaves like before as long as the instructions in the README for executing tests still work.

> The change should also be reflected in the documentation.

I consider this file an implementation detail and would rather encourage people to use the build system targets or prove directly.

Reply:
I implemented the first solution, with a workaround for the coverage statistics.